cpu/nrf5x_common/periph/uart.c

```diff
@@ -112,34 +113,33 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     /* configure TX pin */
     gpio_init(UART_PIN_TX, GPIO_OUT);
     PSEL_TXD = UART_PIN_TX;
-
+    NRF_UART0->PSELRTS = 0xffffffff;        /* pin disconnected */
```

Comment:
nrf52840 variant can have 2 UARTs so I think you should keep the specific else block (nrf52840 != other nrf5x) to set RTS/CTS pin disconnected.

Reply:
I replaced `NRF_UART0` by `dev(uart)`. This way the RTS and CTS will be first disabled and only enabled if needed.

Reply:
Ok I dropped those lines :)